<a href="https://colab.research.google.com/github/Unsinew/The-Start-of-it-all/blob/main/Sales_PredictionsRRR.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [58]:
import pandas as pd
 #Lets grab the data
# load csv from url, display info, and preview rows 
Salesdf = pd.read_csv('/content/drive/MyDrive/2. Week Two/sales_predictions.csv')
Salesdf.info()
Salesdf.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

Looks like everything has loaded

##Inspecting the data

In [59]:
Salesdf.shape

(8523, 12)

Its looking like there is 8523 rows and 12 colums

In [60]:
Salesdf.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

Now we can make sure and see if objects,floats and intergers are in the right places and if there is any discrepancies we can see at a glance.

##Clean Up
How can we clean?
 We can clean('remove') any data that may not be needed.


Lets see if there is any duplicates

In [61]:
Salesdf.duplicated().sum()

0

If there was any we would have been shown form the .duplicated().sum().
What do we do now we can create a filter for the duplicated data so we can remove it.

In [62]:
Sales_du = Salesdf.duplicated(keep = 'last')
## be sure to display it if you need to make sure the changes are as intended
Sales_du[Sales_du]
## and then lets remove it
Dp_Salesdf = Salesdf.drop_duplicates()

Now that we can see that there is no duplicates lets make sure that there is no missing data

In [63]:
Dp_Salesdf.isna().sum().sum()

3873

##Lets see what are the missing values

In [64]:
Dp_Salesdf.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

It looks like there are mising values in Item_Weight and Outlet_Size. 
Now that we know that there is missing values what should we do?
We can either replace the missing values, get rid of the missing values.

##Makeing the call
When deciding what to do with the missing data you need to ask how important is the data. In my call what to do with the missing data isto remove the Item_Weight column and the Outlet_Size because of there large amount of unnessary data and missing data.

In [65]:
Dp_Salesdf.drop(columns = ['Item_Weight', 'Outlet_Size'], inplace = True)
Dp_Salesdf.head()

Item_Identifier Item_Fat_Content  Item_Visibility              Item_Type  \
0           FDA15          Low Fat         0.016047                  Dairy   
1           DRC01          Regular         0.019278            Soft Drinks   
2           FDN15          Low Fat         0.016760                   Meat   
3           FDX07          Regular         0.000000  Fruits and Vegetables   
4           NCD19          Low Fat         0.000000              Household   

   Item_MRP Outlet_Identifier  Outlet_Establishment_Year Outlet_Location_Type  \
0  249.8092            OUT049                       1999               Tier 1   
1   48.2692            OUT018                       2009               Tier 3   
2  141.6180            OUT049                       1999               Tier 1   
3  182.0950            OUT010                       1998               Tier 3   
4   53.8614            OUT013                       1987               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

Lets see if there are more error in the syntax or values

In [66]:
Dp_Salesdf

Item_Identifier Item_Fat_Content  Item_Visibility              Item_Type  \
0              FDA15          Low Fat         0.016047                  Dairy   
1              DRC01          Regular         0.019278            Soft Drinks   
2              FDN15          Low Fat         0.016760                   Meat   
3              FDX07          Regular         0.000000  Fruits and Vegetables   
4              NCD19          Low Fat         0.000000              Household   
...              ...              ...              ...                    ...   
8518           FDF22          Low Fat         0.056783            Snack Foods   
8519           FDS36          Regular         0.046982           Baking Goods   
8520           NCJ29          Low Fat         0.035186     Health and Hygiene   
8521           FDN46          Regular         0.145221            Snack Foods   
8522           DRG01          Low Fat         0.044878            Soft Drinks   

      Item_MRP Outlet_Identifier  Outlet_Establishment_Year  \
0     249.8092            OUT049                       1999   
1      48.2692            OUT018                       2009   
2     141.6180            OUT049                       1999   
3     182.0950            OUT010                       1998   
4      53.8614            OUT013                       1987   
...        ...               ...                        ...   
8518  214.5218            OUT013                       1987   
8519  108.1570            OUT045                       2002   
8520   85.1224            OUT035                       2004   
8521  103.1332            OUT018                       2009   
8522   75.4670            OUT046                       1997   

     Outlet_Location_Type        Outlet_Type  Item_Outlet_Sales  
0                  Tier 1  Supermarket Type1          3735.1380  
1                  Tier 3  Supermarket Type2           443.4228  
2                  Tier 1  Supermarket Type1          2097.2700  
3                  Tier 3      Grocery Store           732.3800  
4                  Tier 3  Supermarket Type1           994.7052  
...                   ...                ...                ...  
8518               Tier 3  Supermarket Type1          2778.3834  
8519               Tier 2  Supermarket Type1           549.2850  
8520               Tier 2  Supermarket Type1          1193.1136  
8521               Tier 3  Supermarket Type2          1845.5976  
8522               Tier 1  Supermarket Type1           765.6700  

[8523 rows x 10 columns]